File: frontend/src/lib/types/track.ts
// TypeScript types matching the Go DTOs

export interface TrackDTO {
  id: string;
  sourceId: string;
  sourceType: string;
  title: string;
  artist: string;
  artistId: string;
  album: string;
  albumId: string;
  albumArtist?: string;
  genre?: string;
  year?: number;
  trackNumber?: number;
  discNumber?: number;
  duration: number; // in seconds
  filePath?: string;
  streamUrl?: string;
  format?: string;
  bitRate?: number;
  sampleRate?: number;
  artworkPath?: string;
}

export interface AlbumDTO {
  id: string;
  sourceId: string;
  sourceType: string;
  title: string;
  artist: string;
  artistId: string;
  year?: number;
  genre?: string;
  artworkPath?: string;
  trackCount: number;
  totalDuration: number;
}

export interface ArtistDTO {
  id: string;
  sourceId: string;
  sourceType: string;
  name: string;
  imagePath?: string;
  albumCount: number;
  trackCount: number;
}

export interface PlaylistDTO {
  id: string;
  name: string;
  description?: string;
  trackIds: string[];
  trackCount: number;
  coverPath?: string;
}

export interface SourceDTO {
  id: string;
  name: string;
  type: string;
}

export interface ScanProgressDTO {
  isScanning: boolean;
  totalFiles: number;
  processedFiles: number;
  currentFile: string;
  errors?: string[];
}

File: internal/domain/model/track.go
package model

import "time"

// Track represents a music track from any source
type Track struct {
	// Identification
	ID         string     `json:"id"`
	SourceID   string     `json:"sourceId"`
	SourceType SourceType `json:"sourceType"`

	// Core metadata
	Title       string `json:"title"`
	Album       string `json:"album"`
	AlbumID     string `json:"albumId"`
	Artist      string `json:"artist"`
	ArtistID    string `json:"artistId"`
	AlbumArtist string `json:"albumArtist"`

	// Additional metadata
	Genre       string        `json:"genre"`
	Year        int           `json:"year"`
	TrackNumber int           `json:"trackNumber"`
	DiscNumber  int           `json:"discNumber"`
	Duration    time.Duration `json:"duration"`

	// File-specific (for filesystem sources)
	FilePath   string `json:"filePath,omitempty"`
	FileSize   int64  `json:"fileSize,omitempty"`
	Format     string `json:"format"`           // mp3, flac, m4a, ogg, etc.
	BitRate    int    `json:"bitRate,omitempty"` // kbps
	SampleRate int    `json:"sampleRate,omitempty"` // Hz

	// API-specific (for API sources)
	ExternalID string `json:"externalId,omitempty"`
	StreamURL  string `json:"streamUrl,omitempty"`

	// Artwork
	ArtworkPath string `json:"artworkPath,omitempty"` // Local cache path or URL

	// Timestamps
	AddedAt    time.Time `json:"addedAt"`
	ModifiedAt time.Time `json:"modifiedAt"`
}